src/snowflake/snowpark/modin/pandas/general.py

```diff
@@ -394,26 +394,152 @@ def merge_ordered(
 
 
 @snowpark_pandas_telemetry_standalone_function_decorator
-@pandas_module_level_function_not_implemented()
 @_inherit_docstrings(pandas.merge_asof, apilink="pandas.merge_asof")
 def merge_asof(
     left,
     right,
-    on=None,
-    left_on=None,
-    right_on=None,
+    on: str | None = None,
+    left_on: str | None = None,
+    right_on: str | None = None,
     left_index: bool = False,
     right_index: bool = False,
-    by=None,
-    left_by=None,
-    right_by=None,
-    suffixes=("_x", "_y"),
-    tolerance=None,
+    by: str | list[str] | None = None,
+    left_by: str | None = None,
+    right_by: str | None = None,
+    suffixes: Suffixes = ("_x", "_y"),
+    tolerance: int | Timedelta | None = None,
     allow_exact_matches: bool = True,
     direction: str = "backward",
 ) -> DataFrame:  # noqa: PR01, RT01, D200
     """
     Perform a merge by key distance.
+
```

Comment:
Any idea why the docstrings for methods in `general.py` are not maintained under the `../docstrings/` directory?

Reply:
There's only about 20 functions, so since it's a small enough set having the docstrings for each one doesn't bloat the file too much. But for uniformity, I can refactor this in a follow-up PR. I'll file a JIRA for it.

Reply:
I think it's more about minimizing the conflicts with Modin OSS. Thanks!